src/spritecache_internal.h

```diff
@@ -20,15 +20,55 @@
 
 /* These declarations are internal to spritecache but need to be exposed for unit-tests. */
 
+class SpriteDataBuffer {
```

Comment:
nitpick: don't we have an STL container for this? It feels like something that should already exist in the STL :P

Not saying we should use it btw; I am fine with this. I was just looking a bit too deep into this PR, as I was kinda curious, and got dragged away :P

nielsm suggests maybe a `std::vector<char>`?

Reply:
Using std::vector would remove the need for the custom `SpriteDataBuffer` class, at the expense of an extra memory used internally by vector, and time spent initializing the buffer when the vector's size is set.

It would be smaller code though, which is nice.